core/anvil/src/serialization_helper.rs

```diff
@@ -0,0 +1,202 @@
+                    // or in the SeqAccess because we checked the sequence length
+                    let mut element: u16 = ((container.unwrap() >> shift) & 0x1FF) as u16;
+                    shift += 9;
+
+                    if shift >= 64 {
+                        // Take next container
+                        container = seq.next_element()?.map(|x: i64| x as u64);
+
+                        if shift > 64 {
+                            // We have some bits left to get from the next container
+
+                            // same here with the unwrapping
+                            element |= ((container.unwrap() << -(shift - 64 - 9)) & 0x1FF) as u16;
+                        }
+
+                        shift -= 64;
+                    }
+
+                    u9_array.push(element);
+                }
+
+                debug_assert_eq!(container, None);
+                debug_assert_eq!(shift, 0);
+
+                Ok(u9_array)
+            }
+        }
+
+        deserializer.deserialize_seq(PackedVisitor(PhantomData))
+    }
+
+    pub fn serialize<S>(u9_array: &[u16], serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if u9_array.len() % 64 != 0 {
+            // Invalid array length
+            return Err(S::Error::custom("array length must be a multiple of 64"));
+        }
+
+        let packed_iter = (0..u9_array.len() * 9 / 64) // iterate through each resulting u64
+            .map(|i| {
+                (
+                    i / 9 * 64, // u9_array_offset; every 64 u9 the u64 boundary is aligned with the u9 boundary again -> one section. each section is 9 u64 long
+                    i % 9, // container_index; index of the current container in this specific section
+                )
+            })
+            .map(|(u9_array_offset, container_index)| {
+                (0..8) // every u64 (partially) contains 8 u9
+                    .map(|i| {
+                        (
+                            i + container_index * 7 + u9_array_offset, // u9_array index; times 7 because the u9 indices need to overlap
+                            (i as isize) * 9 - container_index as isize, // amount of shift left (negative means shift right)
+                        )
+                    })
+                    .map(|(u9_array_index, shift_left)| {
+                        let u9 = u9_array[u9_array_index] as u64;
+                        if shift_left < 0 {
+                            u9 >> -shift_left as u64
+                        } else {
+                            u9 << shift_left as u64
+                        }
+                    })
+                    .fold(0, |container, u9| container | u9)
+            })
+            .map(|container| container as i64);
+
+        nbt::i64_array(packed_iter, serializer)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use core::iter;
+    use serde::{Deserialize, Serialize};
+    use serde_test::Token;
```

Comment:
Well, not sure what Clippy is on but this import is certainly not unused. Removing it just causes the tests compilation to fail